include/aws/io/socket.h

```diff
@@ -114,6 +114,7 @@ struct aws_socket {
     struct aws_socket_options options;
     struct aws_io_handle io_handle;
     struct aws_event_loop *event_loop;
+    struct aws_channel_handler *handler;
```

Comment:
why!? aws_socket shouldn't need to know about the handler. Is this leftover from a previous iteration?

Reply:
There's no way to go from s_on_socket_write_complete to the handler otherwise